test/Typewriter.Test/TestDataCSharp/com/microsoft/graph/requests/DirectoryObjectReferenceRequest.cs

```diff
@@ -35,48 +35,73 @@ public partial class DirectoryObjectReferenceRequest : BaseRequest, IDirectoryOb
+            return this.SendAsyncWithGraphResponse(null, cancellationToken);
         }
 
         /// <summary>
         /// Puts the specified DirectoryObject reference.
         /// </summary>
         /// <param name="id">The DirectoryObject reference to update.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
         /// <returns>The task to await.</returns>
-        public System.Threading.Tasks.Task PutAsync(string id)
+        public async System.Threading.Tasks.Task PutAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return this.PutAsync(id, CancellationToken.None);
+            var baseUrl = this.Client.BaseUrl;
+            var objectUri = string.Format(@"{0}/users/{1}", baseUrl, id);
+            var payload = string.Empty;
+            using (var stream = new System.IO.MemoryStream())
+            using (var writer = new System.Text.Json.Utf8JsonWriter(stream))
+            {
+                writer.WriteStartObject();
+                writer.WriteString("@odata.id", objectUri);
+                writer.WriteEndObject();
+                await writer.FlushAsync();
+                payload = System.Text.Encoding.UTF8.GetString(stream.ToArray());
+            }
+            this.Method = HttpMethods.PUT;
+            this.ContentType = CoreConstants.MimeTypeNames.Application.Json;
+            await this.SendAsync(payload, cancellationToken).ConfigureAwait(false);
         }
 
         /// <summary>
-        /// Puts the specified DirectoryObject reference.
+        /// Puts the specified DirectoryObject reference and returns <see cref="GraphResponse"/> object.
         /// </summary>
         /// <param name="id">The DirectoryObject reference to update.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
-        /// <returns>The task to await.</returns>
-        public async System.Threading.Tasks.Task PutAsync(string id, CancellationToken cancellationToken)
+        /// <returns>The task to await of <see cref="GraphResponse"/>.</returns>
+        public async System.Threading.Tasks.Task<GraphResponse> PutResponseAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
         {
             var baseUrl = this.Client.BaseUrl;
             var objectUri = string.Format(@"{0}/users/{1}", baseUrl, id);
-            var payload = new Newtonsoft.Json.Linq.JObject(
-                            new Newtonsoft.Json.Linq.JProperty("@odata.id", objectUri));
-            this.Method = "PUT";
-            this.ContentType = "application/json";
-            await this.SendAsync(payload.ToString(), cancellationToken).ConfigureAwait(false);
+            var payload = string.Empty;
+            using (var stream = new System.IO.MemoryStream())
+            using (var writer = new System.Text.Json.Utf8JsonWriter(stream))
```

Comment:
Thanks for pointing this out. This can actually be simplified using [the existing type in core](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/dev/src/Microsoft.Graph.Core/Models/ReferenceRequestBody.cs) to 

```cs
this.Method = HttpMethods.PUT;
this.ContentType = CoreConstants.MimeTypeNames.Application.Json;
var referenceRequestBody = new ReferenceRequestBody()
{
    ODataId = string.Format(@"{0}/users/{1}", this.Client.BaseUrl, id)
};
await this.SendAsync(referenceRequestBody, cancellationToken).ConfigureAwait(false);
```

Will resolve this in the next update to #562

Reply:
Resolved via #562